src/ceph-disk/ceph_disk/main.py

```diff
@@ -303,7 +303,12 @@ class Error(Exception):
 
     def __str__(self):
         doc = _bytes2str(self.__doc__.strip())
-        return ': '.join([doc] + [_bytes2str(a) for a in self.args])
+        try:
+            str_type = basestring
+        except NameError:
+            str_type = str
+        args = [a if isinstance(a, str_type) else str(a) for a in self.args]
```

Comment:
@alfredodeza in the case of python2, a `bytes` instance is a `basestring`, so does `str` and `unicode`. in other words, `b'hello'`, `'hello'`, and `u'hello'` are all "`basestring`". the reason i want to check if `a` is an instance of `str_type` is to see if we need to convert it to a `str` (non-unicode) on python2 or a `str` (unicode) on python3 by calling the magic method of `__str__()` or `__repr__()` magic method. here, `str` is the class we can use it to convert an printable object to its string representation. on python3, there is no more non-unicode `str` and unicode `str`, and  `str` is unified. the reason i check if `isinstance(a, basestring)` is to see if we need to convert it to a `str`. and `basestring` type **cannot** be instantiated. instead it is just used to tell if an object is a "string" in general sense. please note, normally `__str__()` does not return a `bytes` instance. so we can't use `str_type` on python2.